BUILD

```diff
@@ -7247,6 +7247,28 @@ grpc_cc_library(
     ],
 )
 
+# This is an EXPERIMENTAL target subject to change.
+grpc_cc_library(
+    name = "grpcpp_opencensus_gcp_observability_plugin",
```

Comment:
I know we have called this observability code as plugin in various places but it is not really based on any plugin arch nor is it pluggable AFAICS. Again the name as opencensus in it whereas I thought we are going to hide that

Reply:
As discussed offline: the goal of observability is to hide opencensus (or OTel in future) as implementation details but the C++ app developer is only aware of the grpc observability API.

Reply:
Renamed to `grpcpp_gcp_observability`